crates/ra_proc_macro_srv/src/dylib.rs

```diff
@@ -197,3 +198,23 @@ impl Expander {
             .collect()
     }
 }
+
+#[cfg(windows)]
+fn copy_to_temp_dir(path: &Path) -> io::Result<PathBuf> {
+    let mut to = std::env::temp_dir();
+    let file_name = path.file_name().ok_or_else(|| {
+        io::Error::new(
+            io::ErrorKind::InvalidInput,
+            format!("File path is invalid: {}", path.display()),
+        )
+    })?;
+
+    to.push(file_name);
+    std::fs::copy(path, &to)?;
+    Ok(to)
+}
+
+#[cfg(unix)]
+fn copy_to_temp_dir(path: &Path) -> io::Result<PathBuf> {
```

Comment:
So the function doesn't actually copy the file, maybe the better name would be
`ensure_file_with_lock_free_access()`